core/primitives/src/signature.rs

```diff
@@ -44,6 +44,18 @@ impl PublicKey {
     }
 }
 
+impl SecretKey {
```

Comment:
Regarding `default_signature`. This function is going to be called a lot! We should have the default signature as a const instead of recomputing it each time this function is called.

Reply:
i figured that this would be used for testing.. but now i'm realizing that it seems like we want to instantiate a struct before specifying a valid signature. i can change this for now.